src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1245,17 +1241,19 @@ var ReactCompositeComponent = {
     invariant(inst != null, 'Stateless function components cannot have refs.');
     var publicComponentInstance = component.getPublicInstance();
     if (__DEV__) {
-      var componentName = component && component.getName ?
-        component.getName() : 'a component';
+      let info = '';
+      const ownerName = this.getName();
+      if (ownerName) {
+        info += ' Check the render method of `' + ownerName + '`.';
+      }
+
       warning(
         publicComponentInstance != null ||
-        component._compositeType !== CompositeTypes.StatelessFunctional,
-        'Stateless function components cannot be given refs ' +
-        '(See ref "%s" in %s created by %s). ' +
-        'Attempts to access this ref will fail.',
-        ref,
-        componentName,
-        this.getName()
+        component._compositeType !== ReactCompositeComponentTypes.StatelessFunctional,
+        'Stateless function components cannot be given refs. Attempts to ' +
+        'access this ref will fail.%s%s',
+        info,
+        ReactComponentTreeHook.getStackAddendumByID(this._debugID)
```

Comment:
I think you want `component._debugID`, not `this._debugID` here. Because you want the stack for the *child* component (the one *whose* ref we're trying to attach) rather than the owner (the one who's going to *own* the ref).